ui/src/components/FeedMewWrapper.vue

```diff
@@ -92,4 +98,27 @@ const hideProfile = (index: number) => {
 const keepShowingProfile = (index: number) => {
   window.clearTimeout(profileHideTimeouts.value[index]);
 };
+const toggleLickMew = async () => {
+   // check if already likes
+   if(isLickedByMe()) {
+      // unlick
+      await unlickMew(props.mew.mewEntryHash);
+      // refreshMew();
+   } 
+   else {
+      // lick
+      await lickMew(props.mew.mewEntryHash);
+      // refreshMew();
+   }
+};
+
+const isLickedByMe = () => {
```

Comment:
Is there possibly a way to consult our own source chain for whether a particular Mew is licked by us? Might be faster and reduce dht ops?

Reply:
the zome function `get_full_mew_and_context` which is being used here returns a vec of agent keys that have licked the mew, so no extra dht ops are required to go through the vec and check if an agents key is in that list or not. This seems like a good enough approach for now. There is definitely a more efficient approach that would work better with mews with 1000s or more licks, but I wouldn't be the right person to implement that.